src/cargo/ops/tree/mod.rs

```diff
@@ -386,26 +410,38 @@ fn print_dependencies<'a>(
                 true
             }
         })
+        .filter(|dep| {
+            // Filter out packages to prune.
+            if !pkgs_to_prune.is_empty() {
+                match graph.node(**dep) {
+                    Node::Package { package_id, .. } => {
+                        !pkgs_to_prune.iter().any(|spec| spec.matches(*package_id))
+                    }
+                    _ => true,
+                }
+            } else {
+                true
+            }
```

Comment:
I think it would be fine to skip the empty check. I don't think there is a performance concern here, and removing the check can help simplify a little.

```suggestion
            match graph.node(**dep) {
                Node::Package { package_id, .. } => {
                    !pkgs_to_prune.iter().any(|spec| spec.matches(*package_id))
                }
                _ => true,
            }
```